crates/red_knot_python_semantic/resources/mdtest/annotations/string.md

```diff
@@ -125,7 +125,6 @@ def f7() -> "\x69nt":
 def f8() -> """int""":
     return 1
 
-# error: [annotation-byte-string] "Type expressions cannot use bytes literal"
```

Comment:
Is this not a regression? I think the existing error was correct here -- this looks like a quoted annotation, and the quoted expression is not a valid type annotation

Reply:
> Is this not a regression?

Hm... maybe?

We do emit diagnostics for wrong string-kinds in annotations (all the other examples here). But in general, we do not emit any diagnostic for invalid expressions *inside correct string annotations*. Something like `x: "lambda y: y"` does not lead to a diagnostic on `main`. So having `x: "b'int'" *not* emit a diagnostic either seems consistent to me.

If we generally want to show errors for all these cases, we can also do that, I believe.

Reply:
I see. I think we probably _do_ want errors for these cases? But if this is consistent with what we have currently elsewhere, then it's probably better to leave it as you have it for now, and tackle it in isolation as a followup.